core/src/toga/app.py

```diff
@@ -485,24 +473,24 @@ def __init__(
 
         self.on_exit = on_exit
 
-        # We need the command set to exist so that startup et al can add commands;
+        # We need the command set to exist so that startup et al. can add commands;
         # but we don't have an impl yet, so we can't set the on_change handler
         self._commands = CommandSet()
 
         self._startup_method = startup
 
-        self._main_window = None
+        self._main_window: MainWindow | None = None
         self._windows = WindowSet(self)
 
-        self._full_screen_windows = None
+        self._full_screen_windows: tuple[Window, ...] | None = None
 
         self._create_impl()
 
         # Now that we have an impl, set the on_change handler for commands
         self.commands.on_change = self._impl.create_menus
 
-    def _create_impl(self):
-        return self.factory.App(interface=self)
+    def _create_impl(self) -> None:
+        self.factory.App(interface=self)
```

Comment:
All that's an interesting oddity... I've never noticed that we're not actually using the result of the factory call, relying on the factory to set the attribute on the interface.

Not something we need to fix here, but I'll remember to clean this up when I'm landing the #2244 changes.